File: clojure/leap/src/leap.clj
(ns leap)

(def leap-year-rules [[400 true] [100 false] [4 true]])

(defn match-rules [year [interval is-leap]]
  (zero? (mod year interval)))

(defn leap-year? [year]
  (or
   (second (first (filter (partial match-rules year) leap-year-rules)))
   false))

File: cpp/leap/leap.cpp
#include "leap.h"

namespace leap {
    bool is_leap_year(int const year) {
	auto divisible_by =
	    [=] (int const divisor) { return year % divisor == 0; };

	return divisible_by(400) || divisible_by(4) && !divisible_by(100);
    }
}
